strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-nuget-layout-provider/src/test/java/org/carlspring/strongbox/cron/jobs/RegenerateNugetChecksumCronJobTestIT.java

```diff
@@ -36,343 +41,277 @@
+        Path nupkgSha512Path = Paths.get(artifactNupkgPath.toString() + ".sha512");
+        deleteIfExists(nupkgSha512Path);
+        assertTrue(Files.notExists(nupkgSha512Path),"The checksum file for nupkg artifact exist!");
 
-        deleteIfExists(new File(artifactPath, "/1.0.0/org.carlspring.strongbox.checksum-second.1.0.0.nupkg.sha512"));
-        deleteIfExists(new File(artifactPath, "/1.0.0/org.carlspring.strongbox.checksum-second.nuspec.sha512"));
+        Path nuspecSha512Path = Paths.get(artifactNuspecPath.toString() + ".sha512");
+        deleteIfExists(nuspecSha512Path);
+        assertTrue(Files.notExists(nuspecSha512Path),"The checksum file for nuspec artifact exist!");
 
-        assertFalse(
-                new File(artifactPath, "/1.0.0/org.carlspring.strongbox.checksum-second.1.0.0.nupkg.sha512").exists(),
-                "The checksum file for artifact exist!");
-
-        List<File> resultList = new ArrayList<>();
+        List<Path> resultList = new ArrayList<>();
         jobManager.registerExecutionListener(jobKey.toString(), (jobKey1, statusExecuted) ->
         {
             if (!StringUtils.equals(jobKey1, jobKey.toString()) || !statusExecuted)
             {
                 return;
             }
-            resultList.add(new File(artifactPath,
-                                    "/1.0.0/org.carlspring.strongbox.checksum-second.1.0.0.nupkg.sha512"));
-            resultList.add(new File(artifactPath,
-                                    "/1.0.0/org.carlspring.strongbox.checksum-second.nuspec.sha512"));
+            resultList.add(nupkgSha512Path);
+            resultList.add(nuspecSha512Path);
         });
 
-        addCronJobConfig(jobKey, jobName, RegenerateChecksumCronJob.class, STORAGE1, REPOSITORY_RELEASES,
+        addCronJobConfig(jobKey,
+                         jobName,
+                         RegenerateChecksumCronJob.class,
+                         repository.getStorage().getId(),
+                         repository.getId(),
                          properties ->
                          {
                              properties.put("basePath", "org.carlspring.strongbox.checksum-second");
                              properties.put("forceRegeneration", "false");
                          });
 
-        assertTrue(expectEvent(), "Failed to execute task!");
+        await().atMost(EVENT_TIMEOUT_SECONDS, TimeUnit.SECONDS).untilTrue(receivedExpectedEvent());
 
         assertEquals(2, resultList.size());
-        resultList.forEach(f -> {
-            assertTrue(f.exists(),
-                       "The checksum file " + f.toString() + " doesn't exist!");
-            assertTrue(f.length() > 0,
-                       "The checksum file is empty!");
+        resultList.forEach(path -> {
+            assertTrue(Files.exists(path),
+                       "The checksum file " + path.toString() + " doesn't exist!");
+            try
+            {
+                assertTrue(Files.size(path) > 0,
```

Comment:
@ptirador could you please use `ThrowingConsumer` here.

Reply:
Could you please tell me how that works?

Reply:
try it like this:
```
resultList.forEach(ThrowingConsumer.unchecked(path -> {
            assertTrue(Files.exists(path), "The checksum file " + path.toString() + " doesn't exist!");
            assertTrue(Files.size(path) > 0, "The checksum file is empty!");
        }));
```